pkg/provider/provider.go

```diff
@@ -849,8 +852,10 @@ func (k *kubeProvider) Create(
 		initialized = partialErr.Object()
 	}
 
+	obj := checkpointObject(newInputs, initialized, newResInputs)
```

Comment:
Feels like we should update `checkpointObject` to take the `initialApiVersion` as well.